test/e2e/ingress/pathtype_exact.go

```diff
@@ -0,0 +1,117 @@
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("[Ingress] [PathType] exact", func() {
+	f := framework.NewDefaultFramework("exact")
+
+	ginkgo.BeforeEach(func() {
+		f.NewEchoDeployment()
+	})
+
+	ginkgo.It("should choose exact location for /exact", func() {
+		if !f.IsIngressV1Ready {
+			ginkgo.Skip("Test requires Kubernetes v1.18 or higher")
+		}
+
+		host := "exact.path"
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/configuration-snippet": `more_set_input_headers "pathType: exact";`,
+		}
+
+		var exactPathType = networkingv1beta1.PathTypeExact
+		ing := framework.NewSingleIngress("exact", "/exact", host, f.Namespace, framework.EchoService, 80, annotations)
+		ing.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].PathType = &exactPathType
+		f.EnsureIngress(ing)
+
+		annotations = map[string]string{
+			"nginx.ingress.kubernetes.io/configuration-snippet": `more_set_input_headers "pathType: prefix";`,
+		}
+
+		ing = framework.NewSingleIngress("exact-sufix", "/exact", host, f.Namespace, framework.EchoService, 80, annotations)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, host) &&
+					strings.Contains(server, "location = /exact") &&
+					strings.Contains(server, "location /exact")
+			})
+
+		body := f.HTTPTestClient().
+			GET("/exact").
+			WithHeader("Host", host).
+			Expect().
+			Status(http.StatusOK).
+			Body().
+			Raw()
+
+		assert.NotContains(ginkgo.GinkgoT(), body, "pathtype=prefix")
+		assert.Contains(ginkgo.GinkgoT(), body, "pathtype=exact")
+
+		body = f.HTTPTestClient().
+			GET("/exact/sufix").
+			WithHeader("Host", host).
+			Expect().
+			Status(http.StatusOK).
+			Body().
+			Raw()
+
+		assert.Contains(ginkgo.GinkgoT(), body, "pathtype=prefix")
+
+		annotations = map[string]string{
+			"nginx.ingress.kubernetes.io/configuration-snippet": `
+				more_set_input_headers "pathType: prefix";
+				more_set_input_headers "duplicated: true";
+			`,
+		}
+
+		ing = framework.NewSingleIngress("duplicated-prefix", "/exact", host, f.Namespace, framework.EchoService, 80, annotations)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, host) &&
+					strings.Contains(server, "location = /exact") &&
+					strings.Contains(server, "location /exact")
+			})
+
+		body = f.HTTPTestClient().
+			GET("/exact/sufix").
+			WithHeader("Host", host).
+			Expect().
+			Status(http.StatusOK).
+			Body().
+			Raw()
+
+		assert.Contains(ginkgo.GinkgoT(), body, "pathtype=prefix")
+		assert.NotContains(ginkgo.GinkgoT(), body, "pathtype=exact")
+		assert.NotContains(ginkgo.GinkgoT(), body, "duplicated=true")
```

Comment:
Is this to ensure that duplicate ingress created later on do not override the previously created one?

Reply:
yes